crates/ggml/sys/build.rs

```diff
@@ -180,6 +183,24 @@ fn enable_metal(build: &mut cc::Build) {
     build.file("llama-cpp/ggml-metal.m");
     build.flag("-DGGML_USE_METAL");
 
+    println!("cargo:rerun-if-changed=llama-cpp/ggml-metal.metal");
+
+    let manifest_dir_string = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let build_type = env::var("PROFILE").unwrap();
+    // Go up 3 directories from manifest dir to get to root, then go to target/<profile>
+    let path = Path::new(&manifest_dir_string)
+        .ancestors()
+        .nth(3)
+        .unwrap()
+        .join("target")
+        .join(build_type);
+    let output_path = PathBuf::from(path);
+
+    let input_path =
+        Path::new(&env::var("CARGO_MANIFEST_DIR").unwrap()).join("llama-cpp/ggml-metal.metal");
+    let output_path = Path::new(&output_path).join("ggml-metal.metal");
+    std::fs::copy(input_path, output_path).unwrap();
+
```

Comment:
This should not be part of this PR. As you already opened another PR for this change 😅

Reply:
👍  I rebased and took that commit out.